internal/repository/psql/user.go

```diff
@@ -21,10 +21,14 @@ func NewUserRepository(db *sql.DB) *UserRepository {
 func (r *UserRepository) Create(ctx context.Context, input domain.SignUpInput) error {
 	query := "SELECT email FROM users WHERE email=$1"
 	err := r.db.QueryRowContext(ctx, query, input.Email).Scan()
-	if !errors.Is(err, sql.ErrNoRows) {
```

Comment:
this check do it in service method
for example : 
func servicemethod() {
_, err := GetByCredentials(...)
if err == nil {
return domain.ErrUserAlreadyExists....

Reply:
@tarkovskynik to do this check in service I must create additional method which could lookup user only by email or edit current GetByCredentials implementation, is it worth it and which way would I select?